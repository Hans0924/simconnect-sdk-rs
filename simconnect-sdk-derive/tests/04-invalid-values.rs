#![allow(unused_variables, dead_code)]
use simconnect_sdk_derive::SimConnectObject;

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = 123, condition = "none")]
struct GpsData1 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = 123)]
struct GpsData2 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "X")]
struct GpsData3 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "X")]
struct GpsData4 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", interval = "X")]
struct GpsData5 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", interval = 0.0)]
struct GpsData6 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData7 {
    #[simconnect(name = "PLANE LATITUDE", unit = "degrees")]
    pub lat: String,
}

fn main() {}
