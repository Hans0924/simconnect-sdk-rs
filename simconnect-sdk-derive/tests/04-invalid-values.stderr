error: Expected Str, found Int(LitInt { token: 123 })
 --> tests/04-invalid-values.rs:5:14
  |
5 | #[simconnect(period = 123, condition = "none")]
  |              ^^^^^^^^^^^^

error: Expected Str, found Int(LitInt { token: 123 })
 --> tests/04-invalid-values.rs:9:33
  |
9 | #[simconnect(period = "second", condition = 123)]
  |                                 ^^^^^^^^^^^^^^^

error: `period` must be one of ["once", "visual-frame", "sim-frame", "second"]
  --> tests/04-invalid-values.rs:13:14
   |
13 | #[simconnect(period = "X")]
   |              ^^^^^^^^^^^^

error: `condition` must be one of ["none", "changed"]
  --> tests/04-invalid-values.rs:17:33
   |
17 | #[simconnect(period = "second", condition = "X")]
   |                                 ^^^^^^^^^^^^^^^

error: Expected Int, found Str(LitStr { token: "X" })
  --> tests/04-invalid-values.rs:21:33
   |
21 | #[simconnect(period = "second", interval = "X")]
   |                                 ^^^^^^^^^^^^^^

error: Expected Int, found Float(LitFloat { token: 0.0 })
  --> tests/04-invalid-values.rs:25:33
   |
25 | #[simconnect(period = "second", interval = 0.0)]
   |                                 ^^^^^^^^^^^^^^

error: Field type must be one of ["f64", "bool"]
  --> tests/04-invalid-values.rs:31:5
   |
31 | /     #[simconnect(name = "PLANE LATITUDE", unit = "degrees")]
32 | |     pub lat: String,
   | |___________________^
